index.js

```diff
@@ -65,70 +63,355 @@ export class MentionsTextInput extends Component {
     }).start();
   }
 
-  isTriggerDeleted(cursor) {
-    return false;
+  handleDisplaySuggestions(position) {
+      if (!this.triggerMatrix || !this.triggerMatrix.length || !this.isTrackingStarted || this.shouldDeleteTriggerOnBackspace) {
+        return;
+      }
+
+      const keyword = this.getTriggerKeyword(position);
+      const delay = this.props.triggerDelay ? this.props.triggerDelay : 0;
+      if (keyword && keyword.length > delay) {
+        if (this.props.triggerCallback) {
+          this.props.triggerCallback(keyword, this.getSubsequentTriggerIndex(position), this.triggerMatrix);
+        }
+
+        this.openSuggestionsPanel();
+      } else {
+        this.closeSuggestionsPanel();
+      }
+    }
+
+  handleDeleteTriggerOnBackspace(position = 0, index = -2) {
+    if (!this.triggerMatrix || !this.triggerMatrix.length) {
+      return;
+    }
+
+    if (index === -2) {
+      index = this.getSubsequentTriggerIndex(position);
+    }
+
+    const isAtEnd = position === this.state.text.length - 1;
```

Comment:
Do you reuse any of these logic components?

Reply:
there like that for naming/ readability... makes it a lot easier to follow the return value